### DISCUSSION TITLE
use namespace and prefix method in one route group

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1471] [Help]
I have some routes like below

```javascript
Route.group(() => {
    /* start colors Route */
    Route.resource('colors', 'ColorController').apiOnly().formats(['json'])
}).prefix('api').namespace('Api')
```

but when I'm using this method like above I got an error I think in most of the cases `namespace` and `prefix` change with each other .

in this case I should use the `namespace` before each controller and when the routes extends it should be so hard to change or extends

**Comment by rbartholomay** [comment]
you must use th full path:

```javascript
Route.group(() => {
  Route.get('/wsinfo', 'WsController.index')
}).prefix('admin/api/v1')
  .namespace('App/Controllers/Http/admin/api/v1')
```

**Reply by farshadfahimi** [reply]
while using this way you should got an error like below

```
Route.group(...).prefix(...).namespace is not a function
```

**Reply by rbartholomay** [reply]
interesting!?  In my case the code runs fine! But i remember that i have this error also before...

Did you use the latest v5?

**Reply by farshadfahimi** [reply]
no I'm using v4.1

I guess this problem fix in v5
